parsl/jobs/states.py

```diff
@@ -39,12 +39,18 @@ class JobState(IntEnum):
     HELD = 7
     """This job is held/suspended in the batch system"""
 
+    MISSING = 8
+    """Scheduler is no longer tracking this job most likely due to it exiting.
```

Comment:
i feel like this text doesn't adequately describe what `MISSING` is meant to represent. This text also describes jobs that have gone into COMPLETED state because they disappeared from the scheduler and were then set to COMPLETED by the relevant provider, I think... MISSING gives more context to the state beyond it disappearing from the scheduler that is, at least in the htex case, to do with "we know this didn't start properly" and we have a different failure signal that is different from the scheduler.

Reply:
Updated docstring.